bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -992,6 +1078,93 @@ private void handleToneCmd(int current, String channel, Command command, int nbS
         }
     }
 
+    /**
+     * Handle a tcbypass command (only for ASCII protocol)
+     *
+     * @param channel the channel
+     * @param command the received channel command (OnOffType)
+     * @param onCmd the command to be sent to the device to bypass_on
+     * @param offCmd the command to be sent to the device to bypass_off
+     *
+     * @throws RotelException in case of communication error with the device
+     */
+    private void handleTcbypassCmd(String channel, Command command, RotelCommand onCmd, RotelCommand offCmd)
+            throws RotelException, InterruptedException {
+        if (command instanceof OnOffType) {
+            if (command == OnOffType.ON) {
+                connector.sendCommand(onCmd);
+                bass = 0;
+                treble = 0;
+            } else if (command == OnOffType.OFF) {
+                connector.sendCommand(offCmd);
+                Thread.sleep(200);
+                connector.sendCommand(RotelCommand.BASS);
+                Thread.sleep(200);
+                connector.sendCommand(RotelCommand.TREBLE);
+            }
+            Thread.sleep(200);
+            updateChannelState(CHANNEL_BASS);
+            updateChannelState(CHANNEL_TREBLE);
```

Comment:
I propose to move the 2 updateChannelState in the first branch (ON command).
For the second branch (OFF command), this is not required, this will be automatically done as result of the commands "bass?" and "trebble?".

Reply:
fixed